core/drivers/crypto/se050/crypto.mk

```diff
@@ -0,0 +1,70 @@
+ifeq ($(CFG_NXP_SE05X), y)
+# Enable the crypto driver
+CFG_CRYPTO_DRIVER = y
+CFG_CRYPTO_DRIVER_DEBUG ?= 0
+
+# SE050 initialization
+CFG_CORE_SE05X_SCP03_PROVISION ?= n
+CFG_CORE_SE05X_DISPLAY_INFO ?= y
+CFG_CORE_SE05X_SCP03_EARLY ?= y
+CFG_CORE_SE05X_INIT_NVM ?= n
+CFG_CORE_SE05X_OEFID ?= 0
+
+# I2C bus configuration
+CFG_CORE_SE05X_BAUDRATE ?= 3400000
+CFG_CORE_SE05X_I2C_BUS ?= 2
+
+# Extra stacks required to support the Plug and Trust external library
+ifeq ($(shell test $(CFG_STACK_THREAD_EXTRA) -lt 8192; echo $$?), 0)
+CFG_STACK_THREAD_EXTRA = 8192
+endif
+ifeq ($(shell test $(CFG_STACK_TMP_EXTRA) -lt 8192; echo $$?), 0)
+CFG_STACK_TMP_EXTRA = 8192
+endif
```

Comment:
- `CFG_` variables shall be set either with `?=` or `$(call force,...)` (ref: https://optee.readthedocs.io/en/latest/building/gits/optee_os.html#configuration-variables).
-  Drop the `$(shell ...)` thing. 8K is not cast in stone, it's just that it happens to be a value that works with the current P&T code, compiler flags etc. so it is effectively just a reasonable default value. So this block should be:
```make
CFG_STACK_THREAD_EXTRA ?= 8192
CFG_STACK_TMP_EXTRA ?= 8192
```

Reply:
no, that will not work. it will use the default values previously set to 0.

Reply:
Then you need to do it in the proper order, or change the code so that undefined or empty means zero.

Reply:
I dont see how that would help: the SE050 has some minimum requirements in terms of stack driven by the Plug and Trust library (today those requirements are around 8K...in the future hopefully none);  how would you enforce those requirements if not the way I have done it? 

doing what you ask means that if some user, oblivious to the SE050 running on its board, sets the extra stacks config to 2K, his system will not boot....IMO there has to be a comparison and a minimum must be enforced by whoever needs it.

Reply:
?

Reply:
1. You did not address the `=` vs `?=` vs `$(call force,...)` issue. Exercise: try your proposed code with `make CFG_STACK_TMP_EXTRA=0` and comment on what you see ;-)
2. About the check
    > a minimum must be enforced by whoever needs it.

    "Whoever needs it", that's my point. P&T needs the additional stack space, not OP-TEE. OP-TEE has no way to know how much the *external* library needs so how could the value be checked reliably by this Makefile?
If you later improve the library to require less than 8K, you will need to change OP-TEE or waste stack space. At that point, how do you deal with people using the "older" library?

Reply:
@jforissier. Nope. I did address it but the point I made didnt stick.

The code I proposed works as follows
1) if the user requests stack values, the user will get just those values ignoring the SE050 defaults. The user knows better of course.
2) if the user doesnt request stack values and the SE050 is enabled, it will get the default values that the SE050 needs.
3) if the user doesnt request stack values and the SE050 is not enabled, it will get the main defaults (ie 0).

let me know if it matches (or not) your expected use case.

Reply:
> @jforissier. Nope. I did address it but the point I made didnt stick.
> 
> The code I proposed works as follows
> 
> 1) if the user requests stack values, the user will get just those values ignoring the SE050 defaults.

No. Try for instance`CFG_STACK_THREAD_EXTRA=1024 make` and you will get 8192 instead.

>  The user knows better of course.

Indeed. That is why I don't support enforcing a minimal value here.

> 2) if the user doesnt request stack values and the SE050 is enabled, it will get the default values that the SE050 needs.

Agreed.

> 3) if the user doesnt request stack values and the SE050 is not enabled, it will get the main defaults (ie 0).

Agreed.

> let me know if it matches (or not) your expected use case.

I agree with 1-3 and I'm still saying that this is implemented by the following:
```make
ifeq ($(CFG_NXP_SE05X), y)
CFG_STACK_THREAD_EXTRA ?= 8192
CFG_STACK_TMP_EXTRA ?= 8192
endif
```